From 24e875ab6cd24deed32fc85639dc0a4b196d1eed Mon Sep 17 00:00:00 2001
From: Alexandre Rostovtsev <tetromino@gentoo.org>
Date: Thu, 13 Sep 2012 05:32:53 -0400
Subject: [PATCH] core: wait until we daemonized before setting up signals
 (bgo #683932)

If we mask signals before daemonizing, the daemon process will not be
able to handle them, and thus would be unkillable with anything other
than SIGKILL.
---
 src/main.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/main.c b/src/main.c
index 8bc6d3f..edf4b5f 100644
--- a/src/main.c
+++ b/src/main.c
@@ -401,10 +401,6 @@ main (int argc, char *argv[])
 		exit (1);
 	}
 
-	/* Set up unix signal handling */
-	if (!setup_signals ())
-		exit (1);
-
 	/* Set locale to be able to use environment variables */
 	setlocale (LC_ALL, "");
 
@@ -501,6 +497,10 @@ main (int argc, char *argv[])
 			wrote_pidfile = TRUE;
 	}
 
+	/* Set up unix signal handling */
+	if (!setup_signals ())
+		exit (1);
+
 	if (g_fatal_warnings) {
 		GLogLevelFlags fatal_mask;
 
-- 
1.7.12

